tapfreighter/interface.go

```diff
@@ -48,12 +43,15 @@ type CommitmentConstraints struct {
 // String returns the string representation of the commitment constraints.
 func (c *CommitmentConstraints) String() string {
 	var groupKeyBytes, assetIDBytes []byte
-	if c.GroupKey != nil {
-		groupKeyBytes = c.GroupKey.SerializeCompressed()
-	}
-	if c.AssetID != nil {
-		assetIDBytes = c.AssetID[:]
-	}
+
+	c.AssetSpecifier.WhenGroupPubKey(func(groupKey btcec.PublicKey) {
+		groupKeyBytes = groupKey.SerializeCompressed()
+	})
+
+	c.AssetSpecifier.WhenId(func(id asset.ID) {
+		assetIDBytes = id[:]
+	})
+
```

Comment:
Feels like all of this could be a helper method as well, since it's being used at least twice in this commit.
Something like `groupKeyBytes, assetIDBytes := specifier.AsBytes()`.